packages/core/src/base/EngineObject.ts

```diff
@@ -15,6 +15,13 @@ export abstract class EngineObject {
   @ignoreClone
   protected _engine: Engine;
 
+  /**
+   * Get the engine which the object belongs.
+   */
+  get engine(): Engine {
+    return this._engine;
+  }
+
```

Comment:
If you add engine roperty getter here, you should delete `Scene`, `Entity`, `Buffer`’s engine property getter!

Reply:
done